### ISSUE TITLE
add dispersal functions

### ISSUE THREAD

**Issue by goldingn** [goldingn/pop#4]
syntax like:

``` r
dispersal(stage, dispfun)
```

`dispfun` being a special case of a `transfun`, such that they can be combined with transfuns and evaluated both deterministically and stochastically

**Comment by goldingn** [comment]
dispfun should be a `transfun` of type `dispersal`, acting on a `landscape` object (a glorified list of `patch`es) and returning a vector that sums to 1 reflecting the relative probability of moving to that patch.

Given not all individuals should disperse, this `transfun` would typically be multiplied with a probability `transfun`.

I.e.:

``` r
adult_dispersal <- tr(adult ~ adult, p(0.5) * dispfun)
```

The resulting `transfun` should then be recognised as containing a dispersal and handled accordingly (similarly to a rate), so that the selected adults are distributed into new patches.

**Comment by goldingn** [comment]
The above definition requires tweaking of `pop:::expected()` so that it acts on landscapes, not patches, and of the use-defined `transfun`s so that they are vectorised.

E.g. the readme example would become:

``` r
ddfun <- function (landscape) {
  adult_density <- population(landscape, 'adults') / area(landscape)
  0.9 * exp(-adult_density / 1000)
}
```

where `population(landscape, 'adults')` and `area(landscape)` return vectors, so `ddfun` does too.

Changes will also be needed in pop:::`as.matrix.dynamic()`, and in `pop:::update()` to handle matrix population states (rows being patches)

**Comment by goldingn** [comment]
closed in #30
